### PR TITLE
fix: Changed dynamic to group meeting and udpated meta

### LINKED ISSUE #13019: [CAL-2862] change "Dynamic" to "Group Meeting" and fix Meta and OG images
before:

image

After:

Title: Meet Cécile Veneziani & Nicolas Filzi
OG Image: "Group meeting"

add Description: Meet both Cécile Veneziani and Nicolas Filzi in a conference call.

also the booking page:

CleanShot 2024-01-03 at 16 59 06@2x

if it's more than two people, it'll be:

* 3 people: Bailey Pumfleet, Peer Richelsen and John Doe.
* more than 3 people: Bailey Pumfleet, Peer Richelsen and 3 others

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2862](https://linear.app/calcom/issue/CAL-2862/change-dynamic-to-group-meeting-and-fix-meta-and-og-images)</sub>

### PR DESCRIPTION
**Pull request by Amit91848**
## What does this PR do?

Fixes #13019

Screenshot (6)

Screenshot (7)

Screenshot (8)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (Unit/Integration/E2E or any other test)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/bookings/components/BookerSeo.tsx [hunk 1]
```diff
@@ -38,10 +38,11 @@ export const BookerSeo = (props: BookerSeoProps) => {
   const profileName = event?.profile?.name ?? "";
   const profileImage = event?.profile?.image;
   const title = event?.title ?? "";
+  const description = event?.description ?? "";
   return (
     <HeadSeo
       title={`${rescheduleUid && !!bookingData ? t("reschedule") : ""} ${title} | ${profileName}`}
-      description={`${rescheduleUid ? t("reschedule") : ""} ${title}`}
```
Comment (emrysal) [inline packages/features/bookings/components/BookerSeo.tsx:44]: This looks like it breaks existing event types; the title should remain used I reckon.

Comment (Amit91848) [reply packages/features/bookings/components/BookerSeo.tsx:44]: Should I add a check to use title for normal events and description for dynamic events?

File: packages/lib/defaultEvents.ts [hunk 2]
```diff
@@ -116,8 +116,16 @@ const dynamicEvent = {
...
+export const getDynamicEventDescription = (dynamicUsernames: (string | null)[]): string => {
+  if (dynamicUsernames.length === 2) {
+    return `Meet both ${dynamicUsernames[0]} and ${dynamicUsernames[1]} in a conference call.`;
+  } else if (dynamicUsernames.length === 3) {
+    return `Meet ${dynamicUsernames[0]}, ${dynamicUsernames[1]} and ${dynamicUsernames[2]} in a conference call.`;
+  } else {
+    return `Meet ${dynamicUsernames[0]}, ${dynamicUsernames[1]} and ${
+      dynamicUsernames.length - 2
+    } others in a conference call.`;
+  }
```
Comment (PeerRich) [inline packages/lib/defaultEvents.ts:128]: i wonder if we should use i18n here

File: packages/features/bookings/components/BookerSeo.tsx [hunk 3]
```diff
@@ -38,10 +38,11 @@ export const BookerSeo = (props: BookerSeoProps) => {
...
+      description={`${rescheduleUid ? t("reschedule") : ""} ${description}`}
```
Comment (PeerRich) [inline packages/features/bookings/components/BookerSeo.tsx:45]:
```suggestion
      description={`${rescheduleUid ? t("reschedule") : ""} ${title}`}
```

File: packages/features/bookings/components/BookerSeo.tsx (hunk 1 above)
Comment (PeerRich) [reply packages/features/bookings/components/BookerSeo.tsx]: we can keep title here for now

File: packages/features/eventtypes/lib/getPublicEvent.ts [hunk 4]
```diff
@@ -176,6 +176,7 @@ export const getPublicEvent = async (
 
     return {
       ...defaultEvent,
+      description: getDynamicEventDescription(users.map((user) => user.name)),
```
Comment (PeerRich) [inline packages/features/eventtypes/lib/getPublicEvent.ts:179]:
```suggestion
```

File: packages/features/bookings/components/BookerSeo.tsx [hunk 5]
```diff
@@ -38,6 +38,7 @@ export const BookerSeo = (props: BookerSeoProps) => {
   const profileName = event?.profile?.name ?? "";
   const profileImage = event?.profile?.image;
   const title = event?.title ?? "";
+  const description = event?.description ?? "";
```
Comment (PeerRich) [inline packages/features/bookings/components/BookerSeo.tsx:41]:
```suggestion
```

File: packages/features/eventtypes/lib/getPublicEvent.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ import { getAppFromSlug } from "@calcom/app-store/utils";
 import { getBookingFieldsWithSystemFields } from "@calcom/features/bookings/lib/getBookingFields";
 import { getSlugOrRequestedSlug } from "@calcom/features/ee/organizations/lib/orgDomains";
 import { isRecurringEvent, parseRecurringEvent } from "@calcom/lib";
-import { getDefaultEvent, getUsernameList } from "@calcom/lib/defaultEvents";
+import { getDefaultEvent, getDynamicEventDescription, getUsernameList } from "@calcom/lib/defaultEvents";
```
Comment (PeerRich) [inline packages/features/eventtypes/lib/getPublicEvent.ts:10]:
```suggestion
```

File: packages/lib/defaultEvents.ts [hunk 7]
```diff
@@ -106,7 +106,7 @@ const commons = {
 const dynamicEvent = {
   length: 30,
   slug: "dynamic",
-  title: "Dynamic",
+  title: "Group Meeting",
   eventName: "Dynamic Event",
```
Comment (PeerRich) [inline packages/lib/defaultEvents.ts:110]:
```suggestion
  eventName: "Group Meeting",
```

File: packages/lib/defaultEvents.ts (hunk 2 above)
Comment (PeerRich) [reply packages/lib/defaultEvents.ts:128]: maybe later. not urgent

**Review by PeerRich — APPROVED** [review]

**Review by PeerRich — CHANGES_REQUESTED** [review]
actually now looking at that, i dont like the duplicate names (above and in description)

File: packages/lib/defaultEvents.ts (hunk 2 above)
Comment (PeerRich) [inline packages/lib/defaultEvents.ts:128]:
```suggestion
export const getDynamicEventDescription = (dynamicUsernames: string[], slug: string): string => {
  return `Book a ${slug} min event with ${dynamicUsernames.join(", ")}`;
```

**Review by PeerRich — DISMISSED** [review]

File: packages/lib/defaultEvents.ts [hunk 8]
```diff
@@ -106,8 +106,8 @@ const commons = {
 const dynamicEvent = {
   length: 30,
   slug: "dynamic",
-  title: "Dynamic",
-  eventName: "Dynamic Event",
+  title: "Group Meeting",
+  eventName: "Group Meeting",
   description: "",
```
Comment (PeerRich) [inline packages/lib/defaultEvents.ts:111]:
```suggestion
  description: "Let's have a group meeting with multiple people",
```

File: packages/lib/defaultEvents.ts [hunk 9]
```diff
@@ -106,9 +106,9 @@ const commons = {
 const dynamicEvent = {
   length: 30,
   slug: "dynamic",
-  title: "Dynamic",
-  eventName: "Dynamic Event",
-  description: "",
+  title: "Group Meeting",
+  eventName: "Group Meeting",
+  description: "Let's have a meeting with multiple people",
```
Comment (PeerRich) [inline packages/lib/defaultEvents.ts:111]:
```suggestion
  description: "Join us for a meeting with multiple people",
```

**Review by PeerRich — APPROVED** [review]

**Pull request merged**